codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/factories/AuthMiddlewareFactory.java

```diff
@@ -130,4 +134,16 @@ private static String defaultIdentityResolverForAuthTrait(Trait trait) {
         return "proc { |cfg| cfg[:stub_responses] ? %s.new(proc { %s }) : nil }"
                 .formatted(Hearth.IDENTITY_RESOLVER, identity);
     }
+
+    private static String hearthIdentityForAuthTrait(Trait trait) {
```

Comment:
Can fix this later as part of larger refactoring of codegen to address extensibility of auth schemes (possibly adding the `AuthScheme` interface + an `additionalAuthSchemes` method in interceptors) - but this will need to be changed as well and made non hearth specific.

Reply:
Yeah I think we would have getters on auth scheme that map to hearth classes.